File: src/ru/gurps/generator/desktop/models/characters/CharactersAddon.java
package ru.gurps.generator.desktop.models.characters;

import ru.gurps.generator.desktop.config.Model;

public class CharactersAddon extends Model {
    public Integer id;
    public Integer characterId;
    public Integer featureId;
    public Integer addonId;
    public String cost;
    public String level;

    public CharactersAddon() {
    }

    public CharactersAddon(Integer characterId, Integer featureId, Integer addonId, String cost, String level) {
        this.characterId = characterId;
        this.featureId = featureId;
        this.addonId = addonId;
        this.cost = cost;
        this.level = level;
    }
}

File: src/main/kotlin/ru/gurps/generator/desktop/models/characters/CharactersAddon.kt
package ru.gurps.generator.desktop.models.characters

import ru.gurps.generator.desktop.config.Model
import java.sql.ResultSet

class CharactersAddon : Model {
    override var id: Int = -1
    var characterId: Int = -1
    var featureId: Int = -1
    var itemId: Int = -1
    var cost: Int = 0
    var level: Int = 0

    constructor()
    constructor(result: ResultSet) {
        id = result.getInt("id")
        characterId = result.getInt("characterId")
        featureId = result.getInt("featureId")
        itemId = result.getInt("itemId")
        cost = result.getInt("cost")
        level = result.getInt("level")
    }

    constructor(characterId: Int, featureId: Int, itemId: Int, cost: Int, level: Int) {
        this.characterId = characterId
        this.featureId = featureId
        this.itemId = itemId
        this.cost = cost
        this.level = level
    }
}
